src/components/common/DynamicFields.res

```diff
@@ -98,12 +103,24 @@ module RenderField = {
           )
           switch requiredFieldPath {
           | StringField(stringFieldPath) =>
-            let tempValid = RequiredFieldsTypes.checkIsValid(
-              ~text,
-              ~field_type=required_fields_type.field_type,
-              ~localeObject,
-            )
-
+            let tempValid = switch switch customValidationFunc {
+            | Some(validation) =>
+              validation(
+                ~text,
+                ~field_type=required_fields_type.field_type,
+                ~display_name=Some(required_fields_type.display_name),
+              )
+            | None => OtherValidation
+            } {
+            | Error(errorMessage) => Some(errorMessage)
+            | OtherValidation =>
+              RequiredFieldsTypes.checkIsValid(
+                ~text,
+                ~field_type=required_fields_type.field_type,
+                ~localeObject,
+              )
+            | NoError => None
+            }
```

Comment:
Refactor this if it can be done without a variant.

Reply:
bro, we need this for default validation. or else we need to call the default validation from here only

Reply:
i am working on bacs, sepa and becs. which have same fields like account number but different validations. such that passing a optional function will not be a good idea.